### PR TITLE
New power system added

### PR DESCRIPTION
**Pull request by Grenkin1988**
New power system in place
```Connection``` is used to represent that `Furniture` has connection to power system.
```System``` can contain multiple ```Grid```.
In the future logic for new `Grid` generation should be changed.
In XML – new tag: ```<PowerConnection inputRate="" outputRate="5" capacity=""/>```
`inputRate` – power consumption
`outputRate` – power production
`capacity` – maximum accumulated power
Next step - I am going to add new `Furniture` - accumulator.

### CODE REVIEW

**Comment by Ohda** [comment]
Tested successfully.
I try 4 min maximum speed witout freezing. I think it fixes https://github.com/TeamPorcupine/ProjectPorcupine/issues/796

**Comment by Grenkin1988** [comment]
Merged conflicts after solar panel added.

**Comment by mikejbrown** [comment]
This seems to be running fine except that occasionally I am get spammed by NullReferenceExceptions:
```
NullReferenceException: Object reference not set to an instance of an object
InventoryManager.<GetPathToClosestInventoryOfType>m__F (.Inventory i) (at Assets/Scripts/Models/InventoryManager.cs:164)
System.Collections.Generic.List`1[Inventory].TrueForAll (System.Predicate`1 match) (at /Users/builduser/buildslave/mono/build/mcs/class/corlib/System.Collections.Generic/List.cs:608)
InventoryManager.GetPathToClosestInventoryOfType (System.String objectType, .Tile t, Int32 desiredAmount, Boolean canTakeFromStockpile) (at Assets/Scripts/Models/InventoryManager.cs:164)
Character.CheckForJobMaterials () (at Assets/Scripts/Models/Character.cs:703)
Character.Update_DoJob (Single deltaTime) (at Assets/Scripts/Models/Character.cs:575)
Character.Update (Single deltaTime) (at Assets/Scripts/Models/Character.cs:328)
World.UpdateCharacters (Single deltaTime) (at Assets/Scripts/Models/World.cs:152)
WorldController.Update () (at Assets/Scripts/Controllers/WorldController.cs:138)
```
Which seem to be related to a power cell press I placed immediately to the right of the O2 gen, but I'm not able to reproduce it reliably.

**Comment by Grenkin1988** [comment]
@mikejbrown Looks like reported issue is not related to power

**Comment by mikejbrown** [comment]
@Grenkin1988 Yeah, I thought so too. I only reported it because it seemed to be related to the power cell press. But if that's not directly involved in your changes then no worries. :grinning:

**Comment by Grenkin1988** [comment]
@mikejbrown Right now power cell press is not related to power at all :-) It just makes power cells.

**Comment by vogonistic** [comment]
This PR has conflicts.

**Comment by Grenkin1988** [comment]
@vogonistic Resolved conflicts

**Comment by Raketfart** [comment]
Works on my machine, and seems to fix the freezing :+1:

**Comment by Grenkin1988** [comment]
Again conflicts :-(  That's so cool that many new PR merged before old one :-(

**Comment by Ohda** [comment]
@Grenkin1988 I imagine some PR are scarier than others for our gatekeepers.

Can't wait for that one to be added.

**Comment by Mizipzor** [comment]
@Grenkin1988 As a matter of fact @Ohda is correct, if there is a small pull request thats easy to test its usually handled before others. Thats mostly because when I sit down I usually have between 5-10 minutes. If your stuff is more advanced than that it will have to wait until one of my rarer one hour sessions. This is also why getting others to test and vouch for your change is _ absolutely crucial_ if you want bigger things in.

@Raketfart has vouched for this change, and that might be enough if you fix the conflicts. :)

**Comment by Grenkin1988** [comment]
@Mizipzor @vogonistic Resolved

**Comment by Mizipzor** [comment]
I'm gonna go for a leap of faith here.

**Comment by Grenkin1988** [comment]
Great, now everything should be broke and stop working
giphy

File: Assets/Scripts/Models/Furniture.cs [hunk 1]
```diff
@@ -55,8 +56,6 @@ public class Furniture : IXmlSerializable, ISelectable, IContextActionProvider,
     /// </summary>
     private Parameter furnParameters;
 
-    private float powerValue;
```
Comment (mikejbrown) [inline Assets/Scripts/Models/Furniture.cs]: This has broken the power overlay. #1147 I don't know the new power system well enough... what property is to be used in its place?

**Pull request merged**